Enhanced Documentation

int0x81 opened the pull request:
# New Pull Request

For more information, see the `CONTRIBUTING` guide.

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

## Type(s) of Changes

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

This Pull request enhances the documentation according to issue #314 . The documentations for both official supported plugins are now beeing listed under an own section in the "Tooling" tab.

FlorianRappl commented on src/pages/docs/src/codegen/gen-bundlers.js:
```diff
@@ -0,0 +1,28 @@
+const { generateStandardPage } = require('./pages');
+
+function getRoute(name) {
+  return (name && `/bundlers/${name}`) || '';
```
Can we have a `/tooling/...` path here - like for the other sections in tooling?

int0x81 replied:
adapted

FlorianRappl commented on src/pages/docs/src/codegen/gen-bundlers.js:
```diff
@@ -0,0 +1,28 @@
+  const imports = bundlers.map(file => {
+
+    const { mdValue, meta = {} } = render(file, generated);
+    const nameToUse = meta.title.split(' ')[0];
```
I would not use `meta` here (more below).

Instead, the name could be inferred from the path, i.e., just take `piral-cli-webpack` / `piral-cli-parcel` (essentially just the plugin name).

int0x81 replied:
Names are now beeing generated by the file path (e.g. `piral-cli-webpack` / `piral-cli-parcel`).

FlorianRappl commented on src/tooling/piral-cli-parcel/README.md:
```diff
@@ -1,3 +1,7 @@
+---
```
Modifying this MD is not intended;

- it will be inconsistent to the other READMEs
- it will potentially misrender on GitHub / NPM / etc.

Same applies to the `piral-cli-webpack` README

int0x81 replied:
removed the meta tag 'title'

FlorianRappl reviewed (commented):
I think this is good so far.

FlorianRappl commented on src/pages/docs/src/codegen/gen-bundlers.js:
```diff
@@ -0,0 +1,28 @@
+
+    const { mdValue, meta = {} } = render(file, generated);
+    const pathElements = file.split('\\').length;
+    const nameToUse = file.split('\\')[pathElements - 2];
```
This is not a sound solution. The filename is OS dependent. Either you first normalize to `/` and then split on `/` or you use the OS Separator to perform the splitting.

Also you perform two split operations where only one should be. If you want to avoid a temp. variable just use `filter`:

```ts
const { sep } = require('path');
const [nameToUse] = file.split(sep).filter((_, i, arr) => i === arr.length - 2);
```

int0x81 replied:
adapted

FlorianRappl commented on src/pages/docs/src/codegen/paths.js:
```diff
@@ -109,6 +109,18 @@ function getTools() {
+
+  const toolingRoot = resolve(packagesRoot, 'tooling');
+  const webpackRoot = resolve(resolve(toolingRoot, 'piral-cli-webpack'));
+  const parcelRoot = resolve(resolve(toolingRoot, 'piral-cli-parcel'));
```
Why `resolve(resolve(...))`? JUst a single `resolve` should it.

int0x81 replied:
ups :stuck_out_tongue_winking_eye: , changed to use just one resolve

FlorianRappl commented on src/tooling/piral-cli-parcel/README.md:
```diff
@@ -1,6 +1,4 @@
-[![Piral Logo](https://github.com/smapiot/piral/raw/master/docs/assets/logo.png)](https://piral.io)
```
Why this change?

All our READMEs look similar. Compare:

https://raw.githubusercontent.com/smapiot/piral/develop/src/tooling/piral-cli/README.md
(same for `piral-cli-webpack`)

If the change was made due to the logo etc. being rendered (and looking ugly), please look at how plugins are rendered / done. You can customize this.

int0x81 replied:
rolled back these 2 changes

FlorianRappl reviewed (commented):
I think we are nearly done here.

:beers:

int0x81 commented:
I also changed the README.md documentations for the 2 bundler plugins to be more aligned with the other tooling docs:

- removed the link to piral.io
- removed the Logo above the header

Ideas on that?

FlorianRappl reviewed (approved):
LGTM - thanks :beers:!
